### PR TITLE
Jeff's changes for P2

### PR DESCRIPTION
**Pull request by jadkisson**
Runs without collisions for 200 seconds. Nice job on algorithm, Sean.

1) Moved collision management into process A since it controls movement. Did NOT
move movement to C... too much work. Rather, before any moves occur, run collision
management. Then transmit data to B and C as usual.

2) Created messaging interface to push data across the processes easily. The
arrays were fine for P1, but too much work to figure out what was happening and
were not a uniform interface. Now a Message wraps a set of Positions which is
a collection of Position objects which describes trains.

3) Added ProcessMsgShuttle that can move messages at a specific frequency
between two processes. This will be useful in P3 when messages need to be sent
from process C to control the trains.

4) Changed namespace from com.jeff to com.rtosProject2.

### CODE REVIEW

File: src/main/java/com/rtosProject2/Message.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package com.rtosProject2;
```
Comment (shade34321) [inline src/main/java/com/rtosProject2/Message.java:1]: Yay, my suggestion of passing messages was actually useful!

File: src/main/java/com/rtosProject2/Plane.java [hunk 2]
```diff
@@ -127,9 +127,9 @@ private void SetOccupiedFormat(Label label, boolean isCollision) {
 
                 break;
             case Y:
-                if (!halty && _atRow + 1 == 2)
+                if (!haltY && _atRow + 1 == 2)
                     System.out.println("Y is at 2");
-                if (halty){
+                if (haltY){
```
Comment (shade34321) [inline src/main/java/com/rtosProject2/Plane.java:132]: We could probably put this if first and make the current if and elsif. Might flow a little better for other people reading. We can take care of this in part 3.

File: src/main/java/com/rtosProject2/Gui.java [hunk 3]
```diff
@@ -25,33 +25,36 @@ public static void main(String[] args) throws IOException, InterruptedException
...
         Display display = ConfigureDisplay();
 
-        DoubleBuffer<String[][][]> bufferAB = new DoubleBuffer<>(1);
-        DoubleBuffer<Object[][]> bufferCD = new DoubleBuffer<>(1);
+        DoubleBuffer<Message> bufferAB = new DoubleBuffer<>(1);
+        DoubleBuffer<Message> bufferCD = new DoubleBuffer<>(1);
 
         //TO SINGLE STEP, SET delayMs TO 0.
-        int delayMs = 1000;
+        int delayMs = 5;
```
Comment (shade34321) [inline src/main/java/com/rtosProject2/Gui.java:32]: Why do we have a change in time here?

**Comment by shade34321** [comment]
Overall this looks fine though having process A run collision management seems against the requirements. Though given what we know about him and other people's projects he doesn't really care to much about those requirements.

**Comment by shade34321** [comment]
There is a bug in `ShowMarker` inside `Plane.java`. If I'm reading this correctly we get the current label we need to update, make the marker for said label, set the text inside the label, Set that label to occupied then check some if statement where we set the old row and col to blank. I don't understand that if statement, can you explain it please.

`if (_movement != Movement.Set && _movement != Movement.Random)`

Changing it to this seems to fix it but I don't quite understand the entire method so I'd like your input on it please.

```
    public void ShowMarker(int row, int col, boolean isCollision, String marker) {
        Label label = GetLabelInPlane(row, col);
        marker = String.format(labelFormat, marker.equals("") ? _marker : marker);
        marker = marker.length() > labelFormat.length() ? marker.trim() : marker;
        label.setText(marker);
        //SetOccupiedFormat(label, isCollision);
        if (_movement != Movement.Set && _movement != Movement.Random) {
            SetEmptyFormat(_atRow, _atCol);
            _atRow = row;
            _atCol = col;
        }
        label.setText(marker);
        SetOccupiedFormat(label, isCollision);
    }

File: src/main/java/com/rtosProject2/Gui.java (hunk 3 above)
Comment (jadkisson) [reply src/main/java/com/rtosProject2/Gui.java:32]: Sorry - shouldn't have checked that in. I was running it at 5ms to test it through 200 iterations.

**Pull request merged**